Snakefile

```diff
@@ -0,0 +1,381 @@
+    shell:
+        """
+        ./bin/transform-biosample {input.biosample} \
+            --output {output.biosample}
+        """
+
+rule transform_genbank_data:
+    input:
+        biosample = "data/genbank/biosample.tsv",
+        ndjson = "data/genbank.ndjson"
+    output:
+        fasta = "data/genbank/sequences.fasta",
+        metadata = "data/genbank/metadata_transformed.tsv",
+        duplicate_biosample = "data/genbank/duplicate_biosample.txt"
+    shell:
+        """
+        ./bin/transform-genbank {input.ndjson} \
+            --biosample {input.biosample} \
+            --duplicate-biosample {output.duplicate_biosample} \
+            --output-metadata {output.metadata} \
+            --output-fasta {output.fasta}
+        """
+
+rule transform_gisaid_data:
+    input:
+        ndjson = "data/gisaid.ndjson"
+    output:
+        fasta = "data/gisaid/sequences.fasta",
+        metadata = "data/gisaid/metadata_transformed.tsv",
+        flagged_annotations = temp("data/gisaid/flagged-annotations"),
+        additional_info = "data/gisaid/additional_info.tsv"
+    shell:
+        """
+        ./bin/transform-gisaid {input.ndjson} \
+            --output-metadata {output.metadata} \
+            --output-fasta {output.fasta}  \
+            --output-additional-info {output.additional_info} \
+            --output-unix-newline > {output.flagged_annotations};
+        """
+
+rule download_nextclade:
+    params:
+        dst_source=lambda w: config["s3_dst"] + '/nextclade.tsv.gz',
+        src_source=lambda w: config["s3_src"] + '/nextclade.tsv.gz',
+    output:
+        nextclade = f"data/{database}/nextclade_old.tsv"
+    shell:
+        """
+        ./bin/download-from-s3 {params.dst_source} {output.nextclade} ||  \
+        ./bin/download-from-s3 {params.src_source} {output.nextclade}
+        """
+
+rule get_sequences_without_nextclade_annotations:
+    """Find sequences in FASTA which don't have clades assigned yet"""
+    input:
+        fasta = f"data/{database}/sequences.fasta",
+        nextclade = f"data/{database}/nextclade_old.tsv",
+    output:
+        fasta = f"data/{database}/nextclade.sequences.fasta"
+    shell:
+        """
+        ./bin/filter-fasta \
+            --input_fasta={input.fasta} \
+            --input_tsv={input.nextclade} \
+            --output_fasta={output.fasta} \
+        """
+
+rule annotate_via_nextclade:
+    message:
+        """
+        If there are sequences without clades, then run nextclade to assign
+        clades and calculate other useful metrics.
+        """
+    input:
+        sequences = f"data/{database}/nextclade.sequences.fasta",
+        nextclade_info = f"data/{database}/nextclade_old.tsv"
+    params:
+        new_info = temp(f"data/{database}/nextclade_new.tsv"),
+        nextclade_input_dir = temp(directory(f"data/{database}/nextclade_inputs")),
+        nextclade_output_dir = temp(directory(f"data/{database}/nextclade"))
```

Comment:
I was confused by the use of `temp` here. Since these are not consumed as inputs in another rule, I don't think these ever get removed by snakemake.

Reply:
My understanding is that because they aren't used by another rule they will be cleaned up after this rule (`annotate_via_nextclade`) finishes.

Reply:
Hmm, the [docs](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html?highlight=temp#protected-and-temporary-files) say 
>an output file marked as temp is deleted after all rules that use it as an input are completed

So I wonder if snakemake will delete them if they are `output` rather than `params`?
(These were all included in the [final output](https://github.com/nextstrain/ncov-ingest/runs/4329428003?check_suite_focus=true#step:4:458) from `./bin/clean` so snakemake did not remove them.)

Reply:
Good find! I'll try to fix but might be pushed to a subsequent PR

Reply:
Didn't get to this, added as a to-do in #239